fix MacOS warnings

## Observed Behavior

A few warnings have crept into the MacOS builds.  We should fix these:

```
/Users/runner/work/opencl-intercept-layer/opencl-intercept-layer/intercept/src/intercept.cpp:10290:21: warning: variable 'nameSize' set but not used [-Wunused-but-set-variable]
            size_t  nameSize = 0;
                    ^
/Users/runner/work/opencl-intercept-layer/opencl-intercept-layer/intercept/src/intercept.cpp:10368:21: warning: variable 'nameSize' set but not used [-Wunused-but-set-variable]
            size_t  nameSize = 0;
                    ^
/Users/runner/work/opencl-intercept-layer/opencl-intercept-layer/intercept/src/intercept.cpp:10488:21: warning: variable 'nameSize' set but not used [-Wunused-but-set-variable]
            size_t  nameSize = 0;
                    ^
```

## Desired Behavior

No warnings.

## Steps to Reproduce

See the automated MacOS builds.

--- a/intercept/src/common.h
+++ b/intercept/src/common.h
@@ -50,7 +50,8 @@
         }                       \
     }
 #else
-    #define CLI_ASSERT(x)
+    #include <tuple>
+    #define CLI_ASSERT(x) std::ignore = (x)
 #endif
 
 #if defined(_WIN32) || defined(__linux__) || defined(__FreeBSD__)

